crates/evm/src/query.rs

```diff
@@ -666,28 +682,38 @@ impl<C: sov_modules_api::Context> Evm<C> {
             }
         };
 
-        let mut tx_env = prepare_call_env(&block_env, request.clone())?;
-
         let cfg = self
             .cfg
             .get(working_set)
             .expect("EVM chain config should be set");
-        let mut cfg_env = get_cfg_env(&block_env, cfg, Some(get_cfg_env_template()));
-        cfg_env.disable_block_gas_limit = true;
+        let mut cfg_env = get_cfg_env(&block_env, cfg);
+
+        // set endpoint specific params
+        cfg_env.disable_eip3607 = true;
         cfg_env.disable_base_fee = true;
+        // set higher block gas limit than usual
+        // but still cap it to prevent DoS
+        block_env.gas_limit = 100_000_000;
 
         let mut evm_db = self.get_db(working_set);
 
-        if request.gas.is_none() && tx_env.gas_price > U256::ZERO {
-            // if gas price is set, we need to cap the gas limit
-            cap_tx_gas_limit_with_caller_allowance(&mut evm_db, &mut tx_env)?;
-        }
+        let mut tx_env = prepare_call_env(
+            &block_env,
+            request.clone(),
+            Some(
+                evm_db
+                    .basic(request.from.unwrap_or_default())
+                    .unwrap()
+                    .unwrap_or_default()
+                    .balance,
+            ),
+        )?;
 
         let from = request.from.unwrap_or_default();
         let to = if let Some(to) = request.to {
             to
         } else {
-            let account = evm_db.basic(from).unwrap_or_default();
+            let account = evm_db.basic(from).unwrap();
 
             let nonce = account.unwrap_or_default().nonce;
```

Comment:
Okay, then we should check account.is_some here.

Reply:
if account doesnt exists, the default impl. applies